tcms/telemetry/static/telemetry/js/testing/execution-dashboard.js

```diff
@@ -16,23 +16,43 @@ function preProcessData (data, callbackF) {
 
     jsonRPC('TestCase.filter', { pk__in: caseIds }, function (cases) {
         const testerPerCase = {}
+        const componentsPerCase = {}
         cases.forEach(function (element) {
             testerPerCase[element.id] = element.default_tester__username
         })
 
-        jsonRPC('TestRun.filter', { pk__in: runIds }, function (runs) {
-            const testerPerRun = {}
-            runs.forEach(function (element) {
-                testerPerRun[element.id] = element.default_tester__username
+        jsonRPC('Component.filter', { cases__in: cases.map(i => i.id) }, function (components) {
+            components.forEach(function (component) {
+                if (componentsPerCase[component.cases] === undefined) {
+                    componentsPerCase[component.cases] = []
+                }
+                componentsPerCase[component.cases].push(component.name)
             })
 
-            // augment data set with additional info
-            data.forEach(function (element) {
-                element.default_tester__from_case = testerPerCase[element.case]
-                element.default_tester__from_run = testerPerRun[element.run]
+            jsonRPC('TestRun.filter', { pk__in: runIds }, function (runs) {
+                jsonRPC('Product.filter', { pk__in: runs.map(i => i.build__version__product) }, function (products) {
```

Comment:
This API call contains multiple duplicate values, which for a large enough dataset result in a very large WHERE clause. The majority of these IDs are actually duplicate and need to be removed.

Reply:
This is the only item that needs to be fixed. 

In the majority of cases this will probably map out to a subset of all product, worst case scenario is that every single product will be specified by ID. IMO this is much better than specifying the same IDs hundreds, potentially thousands of times. See lines 12-18 for a example of how duplicate IDs can be removed.